Windows/XinputDevice.cpp

```diff
@@ -234,11 +254,13 @@ int XinputDevice::UpdateState() {
 	for (int i = 0; i < XUSER_MAX_COUNT; i++) {
 		XINPUT_STATE state;
 		ZeroMemory(&state, sizeof(XINPUT_STATE));
+		XINPUT_VIBRATION vibration;
+		ZeroMemory(&vibration, sizeof(XINPUT_VIBRATION));
 		if (check_delay[i]-- > 0)
 			continue;
-		DWORD dwResult = PPSSPP_XInputGetState(i, &state);
+		DWORD dwResult = PPSSPP_XInputGetState(i, &state) && PPSSPP_XInputSetState(i, &vibration);
```

Comment:
Using `&&` is misleading here, because `ERROR_SUCCESS` is actually `0`.  So this means "GetState OR SetState succeed".  I don't recommend using && with integer result codes, probably better to just call them one at a time.

That said, why do we need to set it here?  Seems like we'd always be setting 0 here.  Would it be better to have an `else` inside `ApplyVibration` for `PSP_IsInited()`?

Or we could just let ApplyVibration fill the struct, and then always call SetState after UpdatePad.

Reply:
Not sure if I understood this I just copy&pasted it before mimicking GetState, now moved that to ApplyVibration, but better check that.:3

Reply:
Well, so if you use `GetState() && SetState()`, you get what's called "shortcutting".  It's equivalent to:

```c++
// result = GetState() && SetState();

{
	int temp;
	temp = GetState();
	if (temp != 0) {
		temp = SetState();
	}
	result = temp;
}
```

But, GetState() should almost always return 0 (because "SUCCESS" is 0.)  So it'll never execute `SetState()` if you do `GetState() && SetState()`.  In this case, that was fine, but like I said - it's misleading.

-[Unknown]